### PR TITLE
Leaderboard: hit the API

### PR DESCRIPTION
**Pull request by hydaline-code**
♻ In this part of the project, we made use of API to create our leaderboard game . The following was done : 

✅ Send and receive data from an API.
✅  Refresh button gets all scores for a game created from the API.
✅ Submit button that will save a score for a game created by you (sending data to the API).
✅Made use of arrow functions and used async and await JavaScript features to consume the API.

Kindly review my work thanks .🙏

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,2 +1,94 @@
...
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ name: 'My cool new game' }),
+    });
+    const game = await response.json();
+    // eslint-disable-next-line prefer-destructuring
+    gameId = game.result.split(' ')[3];
+    console.log('Game ID:', gameId);
```
Comment (Ibizugbe) [inline src/index.js]: Kindly take away every `console.log` statement in your codebase as it is unprofessional to push them to your codebase on Github 😉 It is better they are removed after debugging👍🏼.

File: src/index.js [hunk 2]
```diff
@@ -1,2 +1,94 @@
...
+
+const refreshScores = async () => {
+  try {
+    const response = await fetch(`${baseURL}/games/${gameId}/scores`);
+    const scores = await response.json();
+    console.log('Scores:', scores);
+  } catch (error) {
+    console.error('Error fetching scores:', error);
+  }
+};
```
Comment (Ibizugbe) [inline src/index.js]: It is a requirement that when the refresh button is clicked, all scores that have been sent or saved to the API endpoint are retrieved and rendered in the scores list😉. But currently, when we click the refresh button, nothing happens which means it doesn't call the GET API endpoint😉. kindly do well to fix this issue👍🏼.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @hydaline-code 👋🏼👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are correctly setup and they are passing👍🏼.
- [x] Arrow functions have been made use of instead of rehular functions👍🏼.
- [x] gitflow is used👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @hydaline-code,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**